packages/core/src/tech/HlsJsTech.ts

```diff
@@ -75,6 +75,39 @@ export default class HlsJsTech extends BaseTech {
     this.playlistDuration = data?.details?.totalduration;
   }
 
+  getVideoQualities(){
+    if(this.hls){
+      const videoQualities: IVideoQuality[] = [];
+      const levels = this.hls.levels;
```

Comment:
You can optimally just return a mapped array instead `return this.hls.levels.map(...)` ( https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map )

Reply:
Alright. great tip.

Reply:
BTW... any insight as to why `this.hls.currentLevel = level`, freezes the player when `level` is !=-1, for when playing this kind of hls stream ? 😖 
https://cph-p2p-msl.akamaized.net/hls/live/2000341/test/master.m3u8

Reply:
@Nfrederiksen
It seems something is wonky with that stream, not sure what and I don't have time to dig further unfortunetely. But your implementation is fine so it's not that.